### PR TITLE
#4730 - rpath issue python

### LINKED ISSUE #4730: 3.5.0-rc1: RPATH issues with python bindings on mac
Issue overview
--------------

cf https://github.com/NREL/OpenStudio/issues/4722#issuecomment-1294419422

the python bindings libraries are linked to libpython and the rpath isn't set correctly.

```diff
cd /Applications/OpenStudio-3.5.0-rc1/Python
$ otool -L _openstudioairflow.so
_openstudioairflow.so:
	/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 60158.100.133)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	@rpath/libopenstudiolib.dylib (compatibility version 0.0.0, current version 0.0.0)
-	/Users/julien/.pyenv/versions/3.8.12/lib/libpython3.8.dylib (compatibility version 3.8.0, current version 3.8.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)


$ otool -l _openstudioairflow.so | grep -A2 LC_RPATH
268:          cmd LC_RPATH
269-      cmdsize 32
270-         path @loader_path (offset 12)
--
287:          cmd LC_RPATH
288-      cmdsize 40
289-         path @loader_path/../lib/ (offset 12
```

## Possible Solution

* Do we really need to link to libpython?! Ever? At least on mac?
* Otherwise
     * Adjust rpath to include as well `../EnergyPlus` here: https://github.com/NREL/OpenStudio/blob/6a935cab2ae26b4170caf107287df201f4612401/ProjectMacros.cmake#L400-L405
     * add a command that calls `install_name_tool` to make it rpath relative. similar to https://github.com/NREL/OpenStudio/blob/6a935cab2ae26b4170caf107287df201f4612401/python/engine/CMakeLists.txt#L52-L58

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version):
 - Version of OpenStudio (if using an intermediate build, include SHA):

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #4730
 - Trying to get CI to tell me whether it's ok to just not link them.

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: python/module/CMakeLists.txt [hunk 1]
```diff
@@ -6,8 +6,13 @@ foreach(p ${ALL_PYTHON_BINDING_TARGETS})
...
     PRIVATE
     openstudiolib
-    Python::Python
   )
+  if(UNIX AND NOT APPLE)
+    target_link_libraries(${p}
+      PRIVATE
+      Python::Python
+    )
+  endif()
```
Comment (jmarrec) [inline python/module/CMakeLists.txt:15]: Apparently on mac I can get by by NOT linking to libpython for the individual swig bindings (eg _openstudioairflow.so). On Ubuntu I cannot

File: ProjectMacros.cmake [hunk 2]
```diff
@@ -400,14 +400,14 @@ macro(MAKE_SWIG_TARGET NAME SIMPLENAME KEY_I_FILE I_FILES PARENT_TARGET PARENT_S
...
-          INSTALL_RPATH "@loader_path;@loader_path/../lib/"
+          BUILD_RPATH "@loader_path;@loader_path/.."
+          INSTALL_RPATH "@loader_path;@loader_path/../lib"
       )
     elseif(UNIX)
       set_target_properties(${swig_target}
         PROPERTIES
           BUILD_RPATH "$ORIGIN:$ORIGIN/../"
-          INSTALL_RPATH "$ORIGIN:$ORIGIN/../lib/"
+          INSTALL_RPATH "$ORIGIN:$ORIGIN/../lib:$ORIGIN/../EnergyPlus"   # Need to pick up libpython too, or we won't we able to import the python bindings inside system python
```
Comment (jmarrec) [inline ProjectMacros.cmake:410]: We can't import the bindings into the system python if we don't resolve libpython correctly on ubuntu.

File: python/module/CMakeLists.txt (hunk 1 above)
Comment (jmarrec) [reply python/module/CMakeLists.txt:15]: (we should probably figure that out at a later date, fyi @kbenne )

File: ProjectMacros.cmake (hunk 2 above)
Comment (jmarrec) [reply ProjectMacros.cmake:410]: cf my tester here: https://github.com/jmarrec/OpenStudio-Tester/actions/runs/3377012242/jobs/5605431762#step:8:131

And cf after fix: https://github.com/jmarrec/OpenStudio-Tester/actions/runs/3378106286/jobs/5607875849

File: python/module/CMakeLists.txt (hunk 1 above)
Comment (kbenne) [reply python/module/CMakeLists.txt:15]: Is there a linker flag to ignore unresolved symbols somewhere? That would be my first suspicion.

**Review by kbenne — APPROVED** [review]

File: python/module/CMakeLists.txt (hunk 1 above)
Comment (jmarrec) [reply python/module/CMakeLists.txt:15]: https://github.com/NREL/OpenStudio/blob/65bcf102dbde3fe69d83d5ef61ed09694588e925/ProjectMacros.cmake#L380-L381

**Comment by tijcolem** [comment]
Did a full build of this and looks good!

**Review by tijcolem — APPROVED** [review]

**Pull request merged**